### PR TITLE
Split off `unison-runtime` package

### PR DESCRIPTION
**Pull request by ChrisPenner**
## Overview

Split off the runtime for two reasons:

1. parser-typechecker is still a behemoth with too much stuff
2. Moving unison-runtime out means I can tweak runtime things without needing to rebuild ALL of parser-typechecker in `-profile -O2`; which takes ages.

While I was at it, I added `-Wunused-packages` and cleaned that up. There were a lot.

## Implementation notes

* Split off runtime stuff
* Had to move a single definition (`codebaseToCodeLookup`)from `Codebase.hs` into `Codebase.Execute.hs`
* -Wunused-packages doesn't love the "shared" top-level package dependencies, so for packages with tests I switched to specifying dependencies for lib and test separately; which is annoying, but a very small price to pay ultimately.

## Test coverage

Discovered that the `Rsa.hs` tests weren't actually included in any test suite for some reason haha. Added that.

### CODE REVIEW

File: unison-runtime/src/Unison/Codebase/Execute.hs [hunk 1]
```diff
@@ -3,27 +3,36 @@
 --
 -- This allows one to run standalone applications implemented in the Unison
 -- language.
-module Unison.Codebase.Execute where
+module Unison.Codebase.Execute
+  ( execute,
+    codebaseToCodeLookup,
```
Comment (ChrisPenner) [inline unison-runtime/src/Unison/Codebase/Execute.hs:8]: The one definition I had to move

File: parser-typechecker/src/Unison/Codebase.hs [hunk 2]
```diff
@@ -418,12 +415,6 @@ typeLookupForDependencies codebase s = do
             <|> Map.lookup r (TL.effectDecls tl) $> ()
         )
 
-toCodeLookup :: (MonadIO m) => Codebase m Symbol Parser.Ann -> CL.CodeLookup Symbol m Parser.Ann
```
Comment (ChrisPenner) [inline parser-typechecker/src/Unison/Codebase.hs:421]: Moved to Codebase.Execute.codebaseToCodeLookup

File: unison-runtime/tests/Unison/Test/Common.hs
Comment (ChrisPenner) [inline unison-runtime/tests/Unison/Test/Common.hs:1]: I just copy-pasta'd these test utils from parser-typechecker

**Comment by aryairani** [comment]
I poked around a little but couldn't see why things were going wrong nix-wise. Any ideas? cc @sellout

**Comment by aryairani** [comment]
On a second run the error is different or more obvious:

```
       > Configuring benchmark 'relation' for unison-util-relation-0.0.0..
       > building
       > Preprocessing benchmark 'relation' for unison-util-relation-0.0.0..
       > Building benchmark 'relation' for unison-util-relation-0.0.0..
       > [1 of 1] Compiling Main             ( benchmarks/relation/Main.hs, dist/build/relation/relation-tmp/Main.o )
       >
       > benchmarks/relation/Main.hs:8:1: error:
       >     Could not load module ‘Unison.Prelude’
       >     It is a member of the hidden package ‘unison-prelude-0.0.0’.
       >     Perhaps you need to add ‘unison-prelude’ to the build-depends in your .cabal file.
       >     Use -v (or `:set -v` in ghci) to see a list of the files searched for.
       >   |
       > 8 | import Unison.Prelude
       >   | ^^^^^^^^^^^^^^^^^^^^^
       For full logs, run 'nix log /nix/store/kisnjl7rqgpmijdi195h8zhkdqpchcs2-unison-util-relation-bench-relation-0.0.0.drv'.
```

Maybe one dependency deletion too many, @ChrisPenner?

Though I wonder why `stack` didn't complain? Could be something about `hpack` getting run or not getting run and/or an updated file not getting checked in?

**Comment by ChrisPenner** [comment]
@aryairani  Oh interesting; I'm guessing we just don't build benchmarks in CI (which if we're not running them in CI is probably fine, but of course not great if we can accidentally break their builds)

You need to `stack build --bench` to build benchmarks usually, but that also _runs_ the benchmarks.

**Review by aryairani — APPROVED** [review]

**Comment by aryairani** [comment]
> You need to `stack build --bench` to build benchmarks usually, but that also _runs_ the benchmarks.

Need `stack build --bench --no-run-benchmarks` 😅

**Pull request merged**